drivers/usb/udc/udc_stm32.c

```diff
@@ -40,6 +40,20 @@ LOG_MODULE_REGISTER(udc_stm32, CONFIG_UDC_DRIVER_LOG_LEVEL);
 #define UDC_STM32_IRQ		DT_INST_IRQ_BY_NAME(0, UDC_STM32_IRQ_NAME, irq)
 #define UDC_STM32_IRQ_PRI	DT_INST_IRQ_BY_NAME(0, UDC_STM32_IRQ_NAME, priority)
 
+/**
+ * The following defines are used to map the value of the "maxiumum-speed"
+ * DT property to the corresponding definition used by the STM32 HAL.
+ */
+#if defined(CONFIG_SOC_SERIES_STM32L4X)
+#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(USB_OTG_HS_EMB_PHY)
```

Comment:
Something is not right with the 2 lines above. Shouldn't it be:
```c
#if defined(CONFIG_SOC_SERIES_STM32L4X)
/* STM32L4 series USB LL API doesn't provide the 2 below defines */
#define USB_OTG_SPEED_HIGH                     0U
#define USB_OTG_SPEED_HIGH_IN_FULL             1U
#endif

#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(USB_OTG_HS_EMB_PHY)
#define HIGH_SPEED             USB_OTG_SPEED_HIGH_IN_FULL
#else
#define HIGH_SPEED             USB_OTG_SPEED_HIGH
#endif
```
That said, looking at L4 series reference manual, they seem to support only full-speed, not high-speed. The L4 HAL already defines `USB_OTG_SPEED_FULL`. All in one, I think there's no need for a specific case for `CONFIG_SOC_SERIES_STM32L4X` and lines 47 and 53 can be simply removed, AFAICT.

Reply:
Agreed with @etienne-lms above 2 lines are not correct.